src/bsoncxx/include/bsoncxx/v_noabi/bsoncxx/vector/accessor.hpp

```diff
@@ -0,0 +1,343 @@
+        std::is_const<Format>::value,
+        typename format_traits::const_reference,
+        typename format_traits::reference>::type;
+
+    /// Iterator for const-qualified vector elements
+    /// @hideinitializer
+    using const_iterator = typename format_traits::const_iterator;
+
+    /// Element iterator type
+    ///
+    /// @hideinitializer
+    /// For example: std::int8_t*, bsoncxx::v_noabi::vector::elements::float32*,
+    /// bsoncxx::v_noabi::vector::iterators::packed_bit_element
+    using iterator = typename std::conditional<
+        std::is_const<Format>::value,
+        typename format_traits::const_iterator,
+        typename format_traits::iterator>::type;
+
+    /// Type for the underlying byte data
+    ///
+    /// @hideinitializer
+    /// For example: std::uint8_t, std::uint8_t const
+    using byte_type = typename detail::accessor_data<format>::byte_type;
+
+    /// Type for byte counts
+    ///
+    /// @hideinitializer
+    /// For example: std::uint32_t, due to BSON size limits.
+    using byte_count_type = typename detail::accessor_data<format>::byte_count_type;
+
+    /// Type for element counts
+    ///
+    /// @hideinitializer
+    /// For example: std::size_t
+    using element_count_type = typename format_traits::element_count_type;
+
+    /// Type for signed differences between byte iterators
+    ///
+    /// @hideinitializer
+    /// For example: std::ptrdiff_t
+    using byte_difference_type = typename format_traits::byte_difference_type;
+
+    /// Type for signed differences between element iterators
+    ///
+    /// @hideinitializer
+    /// For example: std::ptrdiff_t
+    using element_difference_type = typename format_traits::element_difference_type;
+
+    /// Type for referencing const-qualified vector bytes in-place
+    /// @hideinitializer
+    using const_byte_reference = typename format_traits::const_byte_reference;
+
+    /// Type for referencing vector bytes in-place
+    ///
+    /// @hideinitializer
+    /// For example: std::uint8_t&, std::uint8_t const&, bsoncxx::v_noabi::vector::elements::packed_bit_byte
+    using byte_reference = typename std::conditional<
+        std::is_const<Format>::value,
+        typename format_traits::const_byte_reference,
+        typename format_traits::byte_reference>::type;
+
+    /// Iterator for const-qualified vector bytes
+    /// @hideinitializer
+    using const_byte_iterator = typename format_traits::const_byte_iterator;
+
+    /// Byte iterator type
+    ///
+    /// @hideinitializer
+    /// For example: std::uint8_t*, std::uint8_t const*, bsoncxx::v_noabi::vector::iterators::packed_bit_byte
+    using byte_iterator = typename std::conditional<
+        std::is_const<Format>::value,
+        typename format_traits::const_byte_iterator,
+        typename format_traits::byte_iterator>::type;
+
+    /// @brief Construct a const Vector accessor by validating a bsoncxx::v_noabi::types::b_binary reference.
```

Comment:
```suggestion
    /// @brief Construct a Vector accessor by validating a bsoncxx::v_noabi::types::b_binary reference.
```
Not necessarily const?